src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.digital.ho.hocs.casework.contributions;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.api.SomuItemService;
+import uk.gov.digital.ho.hocs.casework.domain.model.SomuItem;
+import uk.gov.digital.ho.hocs.casework.domain.model.Stage;
+
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+import static uk.gov.digital.ho.hocs.casework.contributions.ContributionStatus.*;
+
+@Service
+@Slf4j
+public class ContributionsProcessorImpl implements ContributionsProcessor {
+
+    public static final String COMPLIANT_CASE_TYPE = "COMP";
+    private final SomuItemService somuItemService;
+
+    @Autowired
```

Comment:
Don't need `@Autowired`, constructor with most parameters is implicitely `@Autowired` while using `@Service`.